### PR TITLE
[azcore] prepping for release

### PR DESCRIPTION
**Pull request by richardpark-msft**
Includes the NewRequestForRequest change, go get -u all, and go mod tidy (also for perf tests)

### CODE REVIEW

File: sdk/azcore/CHANGELOG.md [hunk 1]
```diff
@@ -1,17 +1,11 @@
 # Release History
 
-## 1.12.1 (Unreleased)
+## 1.12.1 (2024-08-06)
```
[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/azcore/CHANGELOG.md:3]: Good point, will update.

[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/azcore/CHANGELOG.md]: Updated.

[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/azcore/CHANGELOG.md]: Oh, I took the wrong date there. I meant to do this this month.

[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/azcore/CHANGELOG.md]: Updated.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by chlowell — APPROVED** [review]

**Pull request merged**